var/spack/repos/builtin/packages/valgrind/package.py

```diff
@@ -69,11 +73,11 @@ def configure_args(self):
         if not (spec.satisfies('%clang') and sys.platform == 'darwin'):
             # Otherwise with (Apple's) clang there is a linker error:
             # clang: error: unknown argument: '-static-libubsan'
-            options.append('--enable-ubsan')
+            if spec.satisfies('+ubsan'):
+                options.append('--enable-ubsan')
+        if spec.satisfies('+only64bit'):
+            options.append('--enable-only64bit')
 
         if sys.platform == 'darwin':
-            options.extend([
-                '--build=amd64-darwin',
-                '--enable-only64bit'
-            ])
+            options.append('--build=amd64-darwin')
```

Comment:
@healther These options were not optional for macOS builds previously. It might have worked with Apple clang but fails for Clang 6.